plugins/modules/dcnm_policy.py

```diff
@@ -732,9 +732,20 @@ def dcnm_policy_get_diff_merge(self):
                         self.diff_create.append(policy)
             elif rc == "DCNM_POLICY_DONT_ADD":
                 # A policy exists and there is no difference between the one that exists and the one that is
-                # is requested to be creted. Check the 'create_additional_policy' flag and crete it if it is
+                # is requested to be created. Check the 'create_additional_policy' flag and create it if it is
                 # set to True
                 if policy["create_additional_policy"] is True:
+
+                    # Check if policy is being created using policy ID. In such a case since we are trying to create
+                    # and additional policy here, clear the "id" and "policyId" fields. Otherwise CREATE will fail
+                    # complaining the policy being created is not unique since a policy with the same policy number
+                    # already exists
+
+                    if "POLICY-" in policy.get("policyId", ''):
+                        policy.pop("id")
+                        policy.pop("policyId")
+                        policy["policy_id_given"] = False
+
                     self.changed_dict[0]["merged"].append(policy)
```

Comment:
Can you add unit tests to cover this fix?

Reply:
Modified existing It and UT to check for this scenario